### ISSUE TITLE
try_ : cannot use local __lambda5 as parameter to non-global template match

### ISSUE THREAD

**Issue by FraMecca** [aliak00/ddash#7]
I have the following function:

```
Event requestUrl(const string url, const string projdir) 
{
    import requests;
    import std.utf : UTFException;

    auto rawdata = () @trusted { return getContent(url).data; } ();

    Event ev = try_!(assumeUTF)(rawdata)
        .match!(
            (string content) => ParseEvent(content, url, projdir),
            (UTFException e) => ToFileEvent(rawdata, url, projdir));                             
    return ev;
}
```

and I get:
```
/home/user/git_mio/scarpa/source/parse.d(66,9): Error: template instance `match!(delegate (string content) => ParseEvent(null, null, null, false).this(content, url, projdir), delegate (UTFException e) => ToFileEvent(null, null, null, false).this(rawdata, url, projdir))` cannot use local __lambda4 as parameter to non-global template match(handlers...)()
/home/user/git_mio/scarpa/source/parse.d(66,9): Error: template instance `match!(delegate (string content) => ParseEvent(null, null, null, false).this(content, url, projdir), delegate (UTFException e) => ToFileEvent(null, null, null, false).this(rawdata, url, projdir))` cannot use local __lambda5 as parameter to non-global template match(handlers...)()
```

I know that this is mainly because more than one context when passing a lambda, but I wonder if a workaround is possible. That would greatly increase the usefulness of try_

**Comment by aliak00** [comment]
Yeah issue 5710. The bane of D's template function alias parameters. I tried `SumType` in a similar situation and it seems to be ok with this. I think because match is global function. So maybe the proper solution (that would partly solve this) is your issue #9. I'll look in to this now.

**Comment by aliak00** [comment]
@FraMecca: Released 0.0.8 now with a work around. I'll close this issue as well and you can ping if the problem is still there.

**Comment by FraMecca** [comment]
Thank you a lot @aliak00
